aten/src/ATen/core/dispatch/DispatchTable.h

```diff
@@ -98,7 +98,10 @@ namespace detail {
     // optimize this if statement away because it's based on
     // a compile time constant.
     if (args_have_tensor_options<Args...>::value) {
-      type_set(TensorOptionsAccumulator().apply(args...).options);
+      TensorOptions tensorOptions = TensorOptionsAccumulator().apply(args...).options;
+      if (tensorOptions.has_dtype() && tensorOptions.has_device() && tensorOptions.has_layout()) {
+        type_set(tensorOptions);
+      }
```

Comment:
Yeah, so replicating the comment I made earlier, I think that we should probably just make TensorOptions return a type set rather than a type id, and then just some configurations of options just result in an empty set being returned. (Or I guess it could return `optional<TypeId>`, I can't think of any configurations where we'd return TWO type ids.)